### PR TITLE
Evolution: Compare symbols before and after

### PR DESCRIPTION
**Pull request by slavapestov**
A resilient change should not *remove* previously-public symbols.

Part of <rdar://problem/40432647>.

### CODE REVIEW

File: lib/AST/Decl.cpp [hunk 1]
```diff
@@ -5335,6 +5335,9 @@ bool AbstractFunctionDecl::isObjCInstanceMethod() const {
 }
 
 static bool requiresNewVTableEntry(const AbstractFunctionDecl *decl) {
+  if (!isa<ClassDecl>(decl->getDeclContext()))
```
Comment (jrose-apple) [inline lib/AST/Decl.cpp]: Do protocols not use this too?

Does this get called on non-class members at all?

File: utils/rth [hunk 2]
```diff
@@ -102,6 +105,36 @@ class ResilienceTest(object):
...
 
+            # Global symbols only, don't display address or type, don't display
+            # undefined symbols
+            nm_cmd = self.target_nm + ["-gjU", output_dylib]
+            verbose_print_command(nm_cmd)
+
+            symbol_file = os.path.join(self.config_dir_map[config],
+                                       self.lib_name + ".symbols")
+
+            returncode = subprocess.call(nm_cmd, stdout=open(symbol_file, "w"))
```
Comment (jrose-apple) [inline utils/rth]: This file handle doesn't get closed, does it?

File: test/SILGen/inline_always.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// RUN: %target-swift-emit-silgen -parse-as-library -enable-sil-ownership %s | tee /tmp/xxx | %FileCheck %s
```
Comment (jrose-apple) [inline test/SILGen/inline_always.swift]: Is this related to the rest of the PR, or did it sneak in?

**Review by jrose-apple — COMMENTED** [review]
Mostly looks good, just a few questions.

File: lib/AST/Decl.cpp (hunk 1 above)
Comment (jrose-apple) [reply lib/AST/Decl.cpp]: Is it possible to make this an assertion instead, then? Or would that mess up call sites too much?

Comment (slavapestov) [reply lib/AST/Decl.cpp]: It messed up a bunch of call sites. I'm not even sure why I included this patch here, it was just something I noticed while looking at the code.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**